modules/packages/Sort.chpl

```diff
@@ -509,7 +509,277 @@ proc insertionSort(Data: [?Dom] ?eltType, comparator:?rec=defaultComparator)
     compilerError("insertionSort() requires 1-D array");
 }
 
+/*
+   Sort the 1D array `Data` in-place using a binary insertion sort algorithm.
+
+   :arg Data: The array to be sorted
+   :type Data: [] `eltType`
+   :arg comparator: :ref:`Comparator <comparators>` record that defines how the
+      data is sorted.
+
+ */
+
+proc binaryInsertionSort(Data: [?Dom] ?eltType, comparator:?rec=defaultComparator) {
+  use Search;
+  chpl_check_comparator(comparator, eltType);
+
+  const low = Dom.low,
+        high = Dom.high,
+        stride = abs(Dom.stride);
+  for i in low..high by stride {
+    var ithVal = Data[i];
+    var inserted=false;
+    var found:bool;
+    var loc:int;
+    var j:int = i-1;
+    (found,loc)=binarySearch(Data,ithVal,comparator=comparator,lo=low,hi=i);
+    while(j>=loc)
+    {
+      Data[j+stride]=Data[j];
+      j-=stride;
+    }
+    Data[j+stride]=ithVal;
+  }        
+}
+
+pragma "no doc"
+/*Error message for multi-dimension arrays */
+proc binaryInsertionSort(Data: [?Dom] ?eltType, comparator:?rec=defaultComparator)
+  where Dom.rank != 1 {
+    compilerError("binaryInsertionSort() requires 1-D array");
+}
+proc timSort(Data: [?Dom] ?eltType, comparator:?rec=defaultComparator) {
+  use Search; 
+  chpl_check_comparator(comparator, eltType);
+  if(Dom.size<=64){
+    binaryInsertionSort(Data,comparator=comparator);
+    return;
+  }
+  const low = Dom.low,
+        high = Dom.high,
+        stride = abs(Dom.stride);
+  //pair for run(start,len)
+  type run = (int,int);
+  //calculate minrun length
+  proc getMinrun(in n: int) : int{
+    var r: int;
+    while(n>=64){
+      r|=n&1;
+      n>>=1;
+      
+    }
+    return n+r;
+  }
+  ///PROCEDURES FOR TIMSORT ONLY///
+  //find runs in the data
+  proc getRuns(minrun:int, Data:[?Dom] ?eltType, comparator:?rec=defaultComparator) {
```

Comment:
Also, I'm OK with these being outside of the scope of `timSort` despite what other sort functions are doing... (I'll probably update those in the future).